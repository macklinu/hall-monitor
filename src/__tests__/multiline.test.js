import hallMonitor from '..'

const rule = context => ({
  ArrowFunctionExpression(node) {
    context.report({
      node,
      message: 'No arrow functions',
    })
  },
})

hallMonitor('multiline', rule, {
  valid: ['function foo() {}'],
  invalid: [
    [
      'const myFunction = () => {',
      '  const a = 1',
      '  const b = 2',
      '  const c = 3',
      '  return a + b + c',
      '}',
    ].join('\n'),
  ],
})
