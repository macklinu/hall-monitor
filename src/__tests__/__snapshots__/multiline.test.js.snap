// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiline invalid const myFunction = () => {
  const a = 1
  const b = 2
  const c = 3
  return a + b + c
} 1`] = `
"const myFunction = () => {
                    ^^^^^^^ [No arrow functions]
  const a = 1
^^^^^^^^^^^^^^^^^^^^^^^^^^^ [No arrow functions]
  const b = 2
^^^^^^^^^^^^^^^^^^^^^^^^^^^ [No arrow functions]
  const c = 3
^^^^^^^^^^^^^^^^^^^^^^^^^^^ [No arrow functions]
  return a + b + c
^^^^^^^^^^^^^^^^^^^^^^^^^^^ [No arrow functions]
}
^^                          [No arrow functions]"
`;

exports[`multiline valid function foo() {} 1`] = `"function foo() {}"`;
